### PR TITLE
Add Algoliasearch js client

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This does not add any live functionality.

If Algolia keys are in place (as part of an optional enhancement for search), then we display some basic console logs that it is live, but it is a WIP.

## Related Tickets & Documents

- Related Issue https://github.com/forem/forem/issues/20833

### CODE REVIEW

File: app/javascript/packs/searchParams.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /* global checkUserLoggedIn, showLoginModal, userData, buildArticleHTML, initializeReadingListIcons */
 /* eslint no-undef: "error" */
 
+// This is a lightweight version of the client, which we should be fine importing regardless of whether it is set up.
+// Could be optimized for optional inclusion in the future.
+const algoliasearch = require('algoliasearch/lite');
```
Comment (maestromac) [inline app/javascript/packs/searchParams.js:6]: non-blocking thought: Weird that their doc suggest this method, the common way to do this is
`import algoliasearch from 'algoliasearch/lite'` This is untested, though.

File: app/javascript/packs/searchParams.js [hunk 2]
```diff
@@ -176,6 +180,12 @@ function search(query, filters, sortBy, sortDirection) {
     }
   });
 
+  // Run Algolia code only if the ID is live.
+  if (document.body.dataset.algoliaId.length > 0) {
```
Comment (maestromac) [inline app/javascript/packs/searchParams.js:184]:
```suggestion
  if (document.body.dataset.algoliaId?.length > 0) {
```
additional safety net if `algoliaId` is undefined for some reason.

File: app/javascript/packs/searchParams.js [hunk 3]
```diff
@@ -205,6 +215,23 @@ function search(query, filters, sortBy, sortDirection) {
     });
 }
 
+function algoliaSearch(searchParams) {
+  console.log('Algolia is a work in progress. You may want to turn it off if you see this message.') /* eslint-disable-line */
+  const env = document.querySelector('meta[name="environment"]').content;
+  const client = algoliasearch('I9R7RVI99F', 'a8977b474b0e2511913ec1af0dd5ea59');
```
Comment (maestromac) [inline app/javascript/packs/searchParams.js:221]: I believe these keys are unintentially left here?

**Review by maestromac — CHANGES_REQUESTED** [review]

File: app/javascript/packs/searchParams.js (hunk 3 above)
Comment (benhalpern) [reply app/javascript/packs/searchParams.js]: Good thing these weren't private keys 😬

**Comment by benhalpern** [comment]
Thanks for review @maestromac, made some changes.

**Review by maestromac — APPROVED** [review]
LGTM

**Pull request merged**